File: 3-Big O/find_nemo2.py
import time


def find_nemo(lst):
    t0 = time.time()

    for i in range(len(lst)):
        if lst[i] == "nemo":
            print("Found NEMO!")

    t1 = time.time()
    print(f"Call to find nemo took {(t1 - t0) * 1000} Milliseconds")


nemo = ["nemo"]
everyone = [
    "dory",
    "bruce",
    "marlin",
    "nemo",
    "gill",
    "bloat",
    "nigel",
    "squirt",
    "darla",
    "hank",
]

large = ["nemo" for i in range(100000)]
find_nemo(large)

File: 3-Big O/find_nemo2.rb
# frozen_string_literal: true

def find_nemo(arr)
  t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)

  arr.each_with_index do |_item, index|
    puts 'Found NEMO!' if arr[index] == 'nemo'
  end

  t1 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
  puts "Call to find Nemo took #{(t1 - t0) * 1000} Milliseconds"
end

nemo = ['nemo']
everyone = %w[dory bruce marlin nemo gill bloat nigel squirt darla hank]
large = Array.new(100_000, 'nemo')

find_nemo(large)
